app/src/sharedTest/java/org/oppia/android/app/onboarding/OnboardingProfileTypeFragmentTest.kt

```diff
@@ -0,0 +1,420 @@
+import org.oppia.android.testing.RunOn
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.TestPlatform
+import org.oppia.android.testing.firebase.TestAuthenticationModule
+import org.oppia.android.testing.junit.InitializeDefaultLocaleRule
+import org.oppia.android.testing.platformparameter.TestPlatformParameterModule
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestCoroutineDispatchers
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.accessibility.AccessibilityTestModule
+import org.oppia.android.util.caching.AssetModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.locale.LocaleProdModule
+import org.oppia.android.util.logging.EventLoggingConfigurationModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.SyncStatusModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.networking.NetworkConnectionDebugUtilModule
+import org.oppia.android.util.networking.NetworkConnectionUtilDebugModule
+import org.oppia.android.util.parser.html.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.image.GlideImageLoaderModule
+import org.oppia.android.util.parser.image.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [OnboardingProfileTypeFragment]. */
+// FunctionName: test names are conventionally named with underscores.
+@Suppress("FunctionName")
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(
+  application = OnboardingProfileTypeFragmentTest.TestApplication::class,
+  qualifiers = "port-xxhdpi"
+)
+class OnboardingProfileTypeFragmentTest {
+  @get:Rule
+  val initializeDefaultLocaleRule = InitializeDefaultLocaleRule()
+
+  @get:Rule
+  val oppiaTestRule = OppiaTestRule()
+
+  @Inject
+  lateinit var testCoroutineDispatchers: TestCoroutineDispatchers
+
+  @Inject
+  lateinit var context: Context
+
+  @Before
+  fun setUp() {
+    Intents.init()
+    setUpTestApplicationComponent()
+    testCoroutineDispatchers.registerIdlingResource()
+  }
+
+  @After
+  fun tearDown() {
+    testCoroutineDispatchers.unregisterIdlingResource()
+    Intents.release()
+  }
+
+  @Test
+  fun testFragment_headerTextIsDisplayed() {
+    launchOnboardingProfileTypeActivity().use {
+      onView(withId(R.id.profile_type_title))
+        .check(matches(isDisplayed()))
+      onView(withId(R.id.profile_type_title))
+        .check(
+          matches(
+            withText(
+              R.string.onboarding_profile_type_activity_header
+            )
+          )
+        )
```

Comment:
```suggestion
      onView(withId(R.id.profile_type_title))
        .check(matches(isDisplayed()))
        .check(
          matches(
            withText(
              R.string.onboarding_profile_type_activity_header
            )
          )
        )
```

Here and other tests where it is checked that the view is displayed and then checked for the content of the view. We can simplify it as shown above.

Reply:
Thanks for the suggestion. I have used the `allOf` check instead as follows:
```kotlin
  onView(withId(R.id.profile_type_title))
        .check(
          matches(
            allOf(
              isDisplayed(),
              withText(
                R.string.onboarding_profile_type_activity_header
              )
            )
          )
        )
```